### ISSUE TITLE
Subreddit suggesting

### ISSUE THREAD

**Issue by XxPKBxX** [Toldry/RedditAutoCrosspostBot#30]
It would be great if `sub_doesnt_exist_bot` also suggests similar subreddit when user types wrong subreddit. For example, when a user types `r/shotposting`, the bot corrects it to `r/shitposting`.

**Comment by Toldry** [comment]
That's a great suggestion and I wanted to implement it before.

it was also discussed in #13

As of now, I do not know of a good way to implement this feature, so the issue will remain open until I figure out a way to do this.

@XxPKBxX you're welcome to take a look at #13 and see if you can think of another way to implement this that might work, the ways that were proposed there did not work well enough.

**Comment by XxPKBxX** [comment]
Maybe you can contact [Reddit List](https://redditlist.com) site owner for their own subreddit databases, and search among them.

**Comment by Toldry** [comment]
Their search function doesn't work well either:

http://redditlist.com/search?searchterm=AskRedit

I deleted one 'd' from 'AskReddit' and redditlist's search doesn't suggest
AskReddit in the results

**Comment by XxPKBxX** [comment]
No, you can request databases of subreddit list and make your own suggesting feature.

**Comment by Toldry** [comment]
but how will I implemented the suggesting feature?

**Comment by XxPKBxX** [comment]
There are ton of libraries that support search suggest feature. I used [this one](https://github.com/lucaong/minisearch) and it worked great.

**Comment by XxPKBxX** [comment]
Oh sorry, I forgot that this bot is made in Python. Try [this one](https://pypi.org/project/fast-autocomplete/).

**Comment by Toldry** [comment]
Neat. I'll take a look at that.

Note to self: read this article later:
https://en.wikipedia.org/wiki/Approximate_string_matching

**Comment by Toldry** [comment]
Update: I found this website that releases an updated .csv file with all subreddits each week

https://frontpagemetrics.com/list-all-subreddits

**Comment by XxPKBxX** [comment]
Nice!
